libraries/PresentableFramework/Presentable.cpp

```diff
@@ -35,6 +35,7 @@ void Presentable::present() {
     // handling - do nothing or show an error code. Let's revisit this later.
     guest = nullptr;
     presentable_state = presenting;
+    presentableWillPerform();
```

Comment:
Since this is an update to the PresentableFramework it is important to note that `present()` also gets called as a means to return control to menus. Code placed in `presentableWillPerform()` may be executed more than once. Perhaps there needs to be some additional separation from a menu and a menu item. Something to think about.